src/Functions/getSetting.cpp

```diff
@@ -58,17 +69,50 @@ class FunctionGetSetting : public IFunction, WithContext
             throw Exception(ErrorCodes::ILLEGAL_COLUMN,
                             "The argument of function {} should be a constant string with the name of a setting",
                             String{name});
+        if (mode == ErrorHandlingMode::Default)
+        {
+            const auto * default_value_column = arguments[1].column.get();
+            if (!default_value_column || default_value_column->size() != 1)
+            {
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                                "The 2nd argument of function {} should be a constant", String{name});
+            }
+        }
 
         std::string_view setting_name{column->getDataAt(0).toView()};
-        return getContext()->getSettingsRef().get(setting_name);
+        Field setting_value;
+        try
+        {
+            setting_value = getContext()->getSettingsRef().get(setting_name);
+        }
+        catch(...)
+        {
+            if (mode == ErrorHandlingMode::Exception)
+                throw;
+            else if (mode == ErrorHandlingMode::Default)
+            {
+                auto default_value_column = arguments[1].column;
+
+                setting_value = (*default_value_column)[0];
+            }
+        }
+        return setting_value;
```

Comment:
You could simplify this whole thing to smth like this:
```
Field setting_value;
if constexpr (mode == ...Exception)
    setting_value = getContext()->...->get(setting_name);
else
{
            const auto * default_value_column = arguments[1].column.get();
            if (!default_value_column || default_value_column->size() != 1)
            {
                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
                                "The 2nd argument of function {} should be a constant", String{name});
            }
            if (!getContext()->...->tryGet(...))
                setting_value = (*default_value_column)[0];   
}
```

Reply:
Ok, I am okay with either style. I have moved the code to above style.

Reply:
It's not only about style, it's about simplicity, deduplication, readability and unnecessary try-catch statements